### ISSUE TITLE
Compiler suggests use of non-existent macro

### ISSUE THREAD

**Issue by jethrogb** [rust-lang/rust#57588]
I'm quite puzzled by the compiler guidance here:

```rust
use std::env;

fn main() {
    env::vav_os("PATH");
}
```

```
error[E0423]: expected function, found macro `env::vav_os`
 --> src/main.rs:4:5
  |
4 |     env::vav_os("PATH");
  |     ^^^^^------
  |     |    |
  |     |    did you mean `var_os`?
  |     did you mean `env::vav_os!(...)`?
```

Why does it think `env::vav_os` is a macro?

**Comment by petrochenkov** [comment]
Duplicate of https://github.com/rust-lang/rust/issues/47361, diagnostics check whether the *first* segment of the path is a macro, and it its (`env!(...)`).

**Comment by petrochenkov** [comment]
Suggestions for macros were written before all the macro modularization reforms.

**Comment by jethrogb** [comment]
This can be closed as a duplicate if the description of #47361 is updated to be more generic.

**Comment by estebank** [comment]
Current output:

```
error[E0423]: expected function, found macro `env::vav_os`
 --> src/main.rs:4:5
  |
4 |     env::vav_os("PATH");
  |     ^^^^^^^^^^^
help: a function with a similar name exists
  |
4 |     env::var_os("PATH");
  |          ^^^^^^
help: use `!` to invoke the macro
  |
4 |     env::vav_os!("PATH");
  |     ^^^^^^^^^^^^
```

Closing as duplicate of #47361.
